ArduCopter/mode.cpp

```diff
@@ -827,3 +827,12 @@ uint16_t Mode::get_pilot_speed_dn()
 {
     return copter.get_pilot_speed_dn();
 }
+
+bool Mode::allows_weathervaning_auto(void) const
```

Comment:
This function seems to be the same as `bool ModeAuto::allows_weathervaning(void) const` should just be able to remove this and swap in the other for the one user.

Reply:
You can remove the ModeAuto function if you want, that is a hang over from previous.  This one has to stay though as take off mode can't see the ModeAuto function, after some other changes to how the take off mission comand is handled.

Reply:
have updated for that to be removed. This meant I needed to change the call in mode_auto.cpp to use Mode::allows_weathervaning_auto

Reply:
Yeah, that's correct 👍